examples/consistency_models/train_consistency_distillation.py

```diff
@@ -0,0 +1,715 @@
+                layers_per_block=2,
+                num_class_embeds=1000,
+                block_out_channels= [32, 64],
+                attention_head_dim=8,
+                down_block_types= [
+                    "ResnetDownsampleBlock2D",
+                    "AttnDownsampleBlock2D",
+                ],
+                up_block_types= [
+                    "AttnUpsampleBlock2D",
+                    "ResnetUpsampleBlock2D",
+                ],
+                resnet_time_scale_shift="scale_shift",
+        
+                )
+        target_model = UNet2DModel(
+                sample_size= args.resolution,
+                in_channels=3,
+                out_channels=3,
+                layers_per_block=2,
+                num_class_embeds=1000,
+                block_out_channels= [32, 64],
+                attention_head_dim=8,
+                down_block_types= [
+                    "ResnetDownsampleBlock2D",
+                    "AttnDownsampleBlock2D",
+                ],
+                up_block_types= [
+                    "AttnUpsampleBlock2D",
+                    "ResnetUpsampleBlock2D",
+                ],
+                resnet_time_scale_shift="scale_shift",
+        
+                )
+    else:
+        config = UNet2DModel.load_config(args.model_config_name_or_path)
+        model = UNet2DModel.from_config(config)
+        target_model = UNet2DModel.from_config(config)
+    
+    # load the model to distill into a consistency model
+    teacher_model = DDPMPipeline.from_pretrained("google/ddpm-cifar10-32").unet
+    model = model.double()
+    target_model = target_model.double() # TODO : support half precision training
+    teacher_model = teacher_model.double()
+    noise_scheduler = CMStochasticIterativeScheduler()
+    num_scales = 40
+    noise_scheduler.set_timesteps(num_scales)
+    timesteps = noise_scheduler.timesteps
+
+
+    # Create EMA for the model, this is the target model in the paper
+    target_model_ema = EMAModel(
+        model.parameters(),
+        decay=args.ema_max_decay,
+        use_ema_warmup=True,
+        inv_gamma=args.ema_inv_gamma,
+        power=args.ema_power,
+        model_cls=UNet2DModel,
+        model_config=model.config,
+    )
+
+    if args.enable_xformers_memory_efficient_attention:
+        if is_xformers_available():
+            import xformers
+
+            xformers_version = version.parse(xformers.__version__)
+            if xformers_version == version.parse("0.0.16"):
+                logger.warn(
+                    "xFormers 0.0.16 cannot be used for training in some GPUs. If you observe problems during training, please update xFormers to at least 0.0.17. See https://huggingface.co/docs/diffusers/main/en/optimization/xformers for more details."
+                )
+            model.enable_xformers_memory_efficient_attention()
+        else:
+            raise ValueError("xformers is not available. Make sure it is installed correctly")
```

Comment:
I wouldn't include xformers in new training scripts, the default pytorch implementations of the kernels should be available

Reply:
Removed!